[Feature] Display active project in navigation drawer

shobhitagarwal1612 opened the pull request:
Closes #230  

- [x]  Highlight selected project
- [x]  Clicking on any project activates it
- [x]  Add `+ Join Project` item
- [x]  Display only joined projects in nav drawer


Screenshot:
![image](https://user-images.githubusercontent.com/8918023/75276476-3806da80-582c-11ea-8836-25a0473d0015.png)

shobhitagarwal1612 commented:
@gino-m Ready for your review. I have made the changes we discussed on hangouts yesterday

gino-m commented on gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java:
```diff
@@ -424,7 +427,7 @@ public boolean onBack() {
   public boolean onNavigationItemSelected(@NonNull MenuItem item) {
     if (item.getGroupId() == R.id.group_join_project) {
-      projectSelectorViewModel.activateProject(item.getOrder());
+      projectSelectorViewModel.activateLocalProject(item.getOrder());
```
We call these "local" here, and "saved" above.  Is there a more consistent name we can use? From the user's perspective, they will "join/leave" a project, which will store them locally for offline use, and sync them when online. Maybe "offline projects" would be clear?

gino-m commented on gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java:
```diff
@@ -131,6 +131,10 @@ public void activateProject(String projectId) {
         .compose(Loadable::loadingOnceAndWrap);
   }
 
+  public Flowable<Loadable<List<Project>>> getSavedProjects() {
```
`getOfflineProjectsOnceAndStream()`, as per https://github.com/google/ground-android/wiki/RxJava-Usage-Guidelines#observable-and-flowable

shobhitagarwal1612 commented:
@gino-m Please take a look now!

shobhitagarwal1612 commented:
Gentle ping...

shobhitagarwal1612 commented:
@gino-m Thanks for your comments!

> When I click "+ Join project", the newly joined project doesn't show up under "Projects". Any idea what could be going wrong?

Fixed.

> Also, note that the "Projects" heading in the menu appears above "Offline areas" and "Sign out" which is obviously incorrect. Could we add a separator above Offline areas to make it clear it's a different section, and perhaps change the heading to say "My projects" to make it clear it's not an exhaustive list of available projects?

Fixed

![image](https://user-images.githubusercontent.com/8918023/76422665-9ce53780-63cb-11ea-93f9-0d940dcc3d83.png)

gino-m commented on gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java:
```diff
@@ -210,6 +204,13 @@ private void addProjectToNavDrawer(List<Project> projects) {
+    // Highlight active project
+    Loadable<Project> activeProject = viewModel.getActiveProject().getValue();
+    if (activeProject != null) {
+      int index = getSelectedProjectIndex(activeProject.value().get());
```
If the project is not yet activated, `value()` will return null here. Instead, you can replace the  `activeProject` null check with `Loadable.getValue(activeProject)` to cover both cases (i.e. Loadable.getValue(activeProject).ifPresent(p => updateSelectedProjectUI(getSelectedProjectIndex(p)));
